### ISSUE TITLE
cmake warnings related to CMP0177 with cmake 3.31.0

### ISSUE THREAD

**Issue by scpeters** [gazebosim/gz-cmake#461]
## Environment
* OS Version: macOS 13.6.3-x86_64
* Source or binary build? source build of gz-plugin3 (https://github.com/gazebosim/gz-plugin/commit/faaf65e805cca46b4539e1941266189beb6b0bae) but appears to affect other versions as well, using binary version of gz-cmake4 4.1.0

## Description
* Expected behavior: it should build with no cmake warnings
* Actual behavior: it has a cmake warning on macOS using the latest cmake 3.31.0 from homebrew

## Steps to reproduce

1. Install cmake 3.31
2. Install gz-plugin dependencies
3. Configure gz-plugin
4. Observe warning

## Output

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gz_plugin-ci-gz-plugin3-homebrew-amd64&build=21)](https://build.osrfoundation.org/job/gz_plugin-ci-gz-plugin3-homebrew-amd64/21/) https://build.osrfoundation.org/job/gz_plugin-ci-gz-plugin3-homebrew-amd64/21/cmake/

~~~
CMake Warning (dev) at /usr/local/share/cmake/gz-cmake4/cmake4/GzInstallAllHeaders.cmake:182 (install):
  Policy CMP0177 is not set: install() DESTINATION paths are normalized.  Run
  "cmake --help-policy CMP0177" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
Call Stack (most recent call first):
  core/include/gz/plugin/CMakeLists.txt:1 (gz_install_all_headers)
~~~

Related: from [cmake 3.31.0 release notes](https://cmake.org/cmake/help/latest/release/3.31.html#other-changes)

> The [install()](https://cmake.org/cmake/help/latest/command/install.html#command:install) command's DESTINATION arguments are now [normalized](https://cmake.org/cmake/help/latest/command/cmake_path.html#normalization), with the exception of INCLUDES DESTINATION arguments in [install(TARGETS)](https://cmake.org/cmake/help/latest/command/install.html#targets). See policy [CMP0177](https://cmake.org/cmake/help/latest/policy/CMP0177.html#policy:CMP0177).

**Comment by scpeters** [comment]
I think we should try setting

~~~
if(POLICY CMP0177)
  cmake_policy(SET CMP0177 NEW)
endif()
~~~

and see if anything breaks

**Comment by azeey** [comment]
cc @j-rivero

**Comment by scpeters** [comment]
> I think we should try setting
> 
> ```
> if(POLICY CMP0177)
>   cmake_policy(SET CMP0177 NEW)
> endif()
> ```
> 
> and see if anything breaks

actually, it may only be caused by the trailing `..` in the destination path on the following line:

* https://github.com/gazebosim/gz-cmake/blob/ceee29190f18847687d1a383c6f6b1219ce95744/cmake/GzInstallAllHeaders.cmake#L184

let's see if we can resolve it more narrowly by normalizing that path

**Comment by scpeters** [comment]
> > I think we should try setting
> > ```
> > if(POLICY CMP0177)
> >   cmake_policy(SET CMP0177 NEW)
> > endif()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and see if anything breaks
> 
> actually, it may only be caused by the trailing `..` in the destination path on the following line:
> 
> * [gz-cmake/cmake/GzInstallAllHeaders.cmake](https://github.com/gazebosim/gz-cmake/blob/ceee29190f18847687d1a383c6f6b1219ce95744/cmake/GzInstallAllHeaders.cmake#L184)
>       
>       
>            Line 184
>         in
>         [ceee291](/gazebosim/gz-cmake/commit/ceee29190f18847687d1a383c6f6b1219ce95744)
>       
>     
>     
>       
>   
>           
>             
>              DESTINATION ${meta_header_install_dir}/..
> 
> let's see if we can resolve it more narrowly by normalizing that path

fix in https://github.com/gazebosim/gz-cmake/pull/467

**Comment by scpeters** [comment]
from https://github.com/gazebosim/gz-cmake/pull/467#issuecomment-2644296274:

> I'll merge this now so it can be included in the next ubuntu nightly build of gz-cmake4, and we can watch the -ci-main- jobs for any failures. If that looks ok, we can proceed with a prerelease to get extra testing with homebrew, or just make a stable release directly

**Comment by scpeters** [comment]
backporting fix to harmonic in https://github.com/gazebosim/gz-cmake/pull/474

**Comment by scpeters** [comment]
> I think we're missing a backport to fortress, right?

correct; the backport will need to be applied manually to [IgnUtils.cmake](https://github.com/gazebosim/gz-cmake/blob/ign-cmake2/cmake/IgnUtils.cmake#L714-L776)

**Comment by scpeters** [comment]
> > I think we're missing a backport to fortress, right?
> 
> correct; the backport will need to be applied manually to [IgnUtils.cmake](https://github.com/gazebosim/gz-cmake/blob/ign-cmake2/cmake/IgnUtils.cmake#L714-L776)

or we could just disable the warning by setting the CMP0177 policy

**Comment by scpeters** [comment]
> > I think we're missing a backport to fortress, right?
> 
> correct; the backport will need to be applied manually to [IgnUtils.cmake](https://github.com/gazebosim/gz-cmake/blob/ign-cmake2/cmake/IgnUtils.cmake#L714-L776)

@Crola1702 backport in https://github.com/gazebosim/gz-cmake/pull/481

it will require a release to fix the warnings in CI
